managedapps/legacyresource/resource.go

```diff
@@ -0,0 +1,194 @@
+package legacyresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	"k8s.io/helm/pkg/helm"
+)
+
+const (
+	defaultNamespace = "default"
+)
+
+type Config struct {
+	HelmClient *helmclient.Client
+	Logger     micrologger.Logger
+
+	Namespace string
+}
+
+type Resource struct {
+	helmClient *helmclient.Client
+	logger     micrologger.Logger
+
+	namespace string
+}
+
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.HelmClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HelmClient must not be empty", config)
+	}
+	if config.Namespace == "" {
+		config.Namespace = defaultNamespace
+	}
+	c := &Resource{
+		helmClient: config.HelmClient,
+		logger:     config.Logger,
+
+		namespace: config.Namespace,
+	}
+
+	return c, nil
+}
+
+func (r *Resource) Delete(name string) error {
+	ctx := context.TODO()
+
+	err := r.helmClient.DeleteRelease(ctx, name, helm.DeletePurge(true))
+	if helmclient.IsReleaseNotFound(err) {
+		return microerror.Maskf(releaseNotFoundError, name)
+	} else if helmclient.IsTillerNotFound(err) {
+		return microerror.Mask(tillerNotFoundError)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) EnsureDeleted(ctx context.Context, name string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring deletion of release %#q", name))
+
+	err := r.helmClient.DeleteRelease(ctx, name, helm.DeletePurge(true))
+	if helmclient.IsReleaseNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q does not exist", name))
+	} else if helmclient.IsTillerNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "warning", "message", "tiller is not found/installed")
+	} else if err != nil {
+		return microerror.Mask(err)
+	} else {
+		r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleted release %#q", name))
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured deletion of release %#q", name))
+
+	return nil
+}
+
+func (r *Resource) Install(name, url, values string, conditions ...func() error) error {
+	ctx := context.TODO()
+
+	tarballPath, err := r.helmClient.PullChartTarball(ctx, url)
```

Comment:
We are pulling chart tarball file from here, and the rest of codes are almost same.